AWSClientRuntime/Sources/Middlewares/Sha256TreeHashMiddleware.swift

```diff
@@ -50,30 +50,19 @@ public struct Sha256TreeHashMiddleware<OperationStackOutput: HttpResponseBinding
               return try await next.handle(context: context, input: input)
           }
     
-    /// Computes the tree-hash and linear hash of a `ByteBuffer`.
+    /// Computes the tree-hash and linear hash of Data.
     /// See http://docs.aws.amazon.com/amazonglacier/latest/dev/checksum-calculations.html for more information.
-    private func computeHashes(bytes: ByteBuffer) -> (String?, String?) {
-        let bufferSize = 1024 * 1024
-        var hashes = [[UInt8]]()
-        
-        while true {
-            var oneMbTempBuffer = ByteBuffer(size: bufferSize)
-            let bytesRead = bytes.readIntoBuffer(buffer: &oneMbTempBuffer)
-            if bytesRead == 0 {
-                break
-            }
-            let hash = oneMbTempBuffer.sha256()
-            hashes.append(hash.toByteArray())
-        }
-        
-        return (bytes.sha256().encodeToHexString(), computeTreeHash(hashes: hashes))
+    private func computeHashes(data: Data) throws -> (String?, String?) {
+        let ONE_MB = 1024 * 1024
+        let hashes: [[UInt8]] = try data.chunked(size: ONE_MB).map{ try $0.sha256().bytes() }
```

Comment:
where is `chunked(size:)` defined?  I don't see it on Foundation.  I think I understand what it does but would like to se the implementation / docs for it

also fix the spacing in `map{` here so SwiftLint is happy

Reply:
It is defined in the new [Data+Extensions](https://github.com/awslabs/aws-sdk-swift/pull/777/files#diff-ad633ba5ccecb6552a23daf2e47879407d56c81a3eae9bcb5201f480a73f7b99R16)